### DISCUSSION TITLE
Any way to prevent macOS from using the menu bar space?

### DISCUSSION THREAD

**Discussion by davidc360** [FelixKratz/SketchyBar#331] [Q&A] [answered]
Since i automatically hide the menu bar, macOS uses that space. For example when i unplug my external monitor, all the windows fill up the entire screen height (due to higher resolution), and my sketchybar is blocked, and i have to resize my windows to make it visible. Is there a way to make macOS not use the sketchybar space? Or is there an app i can use to resize all my windows at once to make them shorter?

**Comment by davidc360** [comment]
One way i found for now: turn off auto hiding menu bar and use topmost=on, caveats: I lose easy access to the menu bar (can still access it by calling a command to hide sketchybar), can't use transparent bar (which looks amazing)

**Comment by FelixKratz** [comment, marked as answer]
I can have a look if I can set this screen area to be occupied, but I doubt that the process has enough permissions to do that, and I think some people want to be able to cover the bar with their windows.

But I think this is something that you can achieve with yabai, even if you dont use any tiling/border features, you can still move and resize all windows on a display connected/removed event. On all other occurrences you can configure it such that you will not even notice that yabai is running and manages your windows. You could probably also hook the window move event and prevent any window from ever covering the bar.

**Reply by davidc360** [reply]
Hey how about setting sketchybar to hide when mouse enters the bar and unhide when leaving? I was looking into using mouse.entered.global but couldnt get it working. Is this currently possible? Below is the code I tried:
```
sketchybar --add item hide left  \
	   --set hide script="sketchybar --bar hidden='off'" \
	   --subscribe hide mouse.entered.global  \
sketchybar --add item unhide left \
	   --set unhide script="sketchybar --bar hidden='off'"  \
	   --subscribe unhide mouse.exited.global  \
```

**Reply by FelixKratz** [reply]
> Hey how about setting sketchybar to hide when mouse enters the bar and unhide when leaving? I was looking into using mouse.entered.global but couldnt get it working. Is this currently possible?

I dont think that this is possible. When the bar is hidden, there will be no mouse events if you leave the region where the bar *would* be if it wasn't hidden. This is not possible without accessibility access. You could however use something like #281 to hide sketchybar on region enter and unhide it on region leave. This requires you to install a global mouse event tap. This would of course also be possible with a c helper process: https://github.com/FelixKratz/SketchyBarHelper

**Reply by davidc360** [reply]
Thanks for the reply. How about hiding on mouse enter, and automatically unhide after X seconds?
```
sketchybar --add item hide left  \
	   --set hide script="sketchybar --bar hidden='on'; sleep 5; sketchybar --bar hidden='off'" \
	   --subscribe hide mouse.entered.global  \
```
Am I using it correctly? It doesn't seem to detect mouse enter events.

**Reply by FelixKratz** [reply]
This works fine for me to hide the bar on mouse enter and unhide it after 5s:
```bash
sketchybar --add item hide left \
           --set hide script="sketchybar --bar hidden=on; sleep 5; sketchybar --bar hidden=off" \
           --subscribe hide mouse.entered.global
```

**Reply by davidc360** [reply]
Awesome!! Thanks
